cpp/velox/memory/VeloxMemoryManager.cc

```diff
@@ -212,21 +212,34 @@ VeloxMemoryManager::VeloxMemoryManager(
 }
 
 namespace {
-MemoryUsageStats collectMemoryUsageStatsInternal(const velox::memory::MemoryPool* pool) {
+MemoryUsageStats collectVeloxMemoryPoolUsageStats(const velox::memory::MemoryPool* pool) {
   MemoryUsageStats stats;
   stats.set_current(pool->currentBytes());
   stats.set_peak(pool->peakBytes());
   // walk down root and all children
   pool->visitChildren([&](velox::memory::MemoryPool* pool) -> bool {
-    stats.mutable_children()->emplace(pool->name(), collectMemoryUsageStatsInternal(pool));
+    stats.mutable_children()->emplace(pool->name(), collectVeloxMemoryPoolUsageStats(pool));
     return true;
   });
   return stats;
 }
+
+MemoryUsageStats collectArrowMemoryPoolUsageStats(const arrow::MemoryPool* pool) {
+  MemoryUsageStats stats;
+  stats.set_current(pool->bytes_allocated());
+  return stats;
+}
 } // namespace
 
 const MemoryUsageStats VeloxMemoryManager::collectMemoryUsageStats() const {
-  return collectMemoryUsageStatsInternal(veloxAggregatePool_.get());
+  const MemoryUsageStats& veloxPoolStats = collectVeloxMemoryPoolUsageStats(veloxAggregatePool_.get());
+  const MemoryUsageStats& arrowPoolStats = collectArrowMemoryPoolUsageStats(arrowPool_.get());
```

Comment:
maybe also make a note which part is using arrow pool, BHJ?

-yuan

Reply:
It seems Arrow mem pool doesn't provide field name like velox mem pool. But it can be inferenced via memory consumer name ```GlutenMemoryConsumer``` just as the PR description.

Reply:
> maybe also make a note which part is using arrow pool, BHJ?

Seems to be that shuffle writer allocates a lot of buffers from Arrow pool.